File: 0009-palindrome-number/0009-palindrome-number.cpp
static const bool init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return true;
}();
class Solution {
public:
    const bool isPalindrome(const int x) {
        const string strX = to_string(x);
        int l = 0, r = strX.size() - 1;
        while (l < r) {
            if (strX[l] != strX[r]) return false;
            ++l;
            --r;
        }
        return true;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0) return false
        val strInt = x.toString()
        var l = 0;
        var r = strInt.length - 1
        while (l < r) {
            if (strInt[l] != strInt[r]) {
                return false
            }
            ++l
            --r
        }
        return true
    }
}